src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -991,4 +1011,86 @@ private static void InsertionSort(Span<TKey> keys, Span<TValue> values)
+            if (typeof(T) == typeof(ushort)) return (ushort)(object)left < (ushort)(object)right ? true : false;
+            if (typeof(T) == typeof(short)) return (short)(object)left < (short)(object)right ? true : false;
+            if (typeof(T) == typeof(uint)) return (uint)(object)left < (uint)(object)right ? true : false;
+            if (typeof(T) == typeof(int)) return (int)(object)left < (int)(object)right ? true : false;
+            if (typeof(T) == typeof(ulong)) return (ulong)(object)left < (ulong)(object)right ? true : false;
+            if (typeof(T) == typeof(long)) return (long)(object)left < (long)(object)right ? true : false;
+            if (typeof(T) == typeof(nuint)) return (nuint)(object)left < (nuint)(object)right ? true : false;
+            if (typeof(T) == typeof(nint)) return (nint)(object)left < (nint)(object)right ? true : false;
+            if (typeof(T) == typeof(float)) return (float)(object)left < (float)(object)right ? true : false;
+            if (typeof(T) == typeof(double)) return (double)(object)left < (double)(object)right ? true : false;
+            return left.CompareTo(right) < 0 ? true : false;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] // compiles to a single comparison or method call
+        public static bool GreaterThan<T>(ref T left, ref T right) where T : IComparable<T>
+        {
+            if (typeof(T) == typeof(byte)) return (byte)(object)left > (byte)(object)right ? true : false;
+            if (typeof(T) == typeof(sbyte)) return (sbyte)(object)left > (sbyte)(object)right ? true : false;
+            if (typeof(T) == typeof(ushort)) return (ushort)(object)left > (ushort)(object)right ? true : false;
+            if (typeof(T) == typeof(short)) return (short)(object)left > (short)(object)right ? true : false;
+            if (typeof(T) == typeof(uint)) return (uint)(object)left > (uint)(object)right ? true : false;
+            if (typeof(T) == typeof(int)) return (int)(object)left > (int)(object)right ? true : false;
+            if (typeof(T) == typeof(ulong)) return (ulong)(object)left > (ulong)(object)right ? true : false;
+            if (typeof(T) == typeof(long)) return (long)(object)left > (long)(object)right ? true : false;
+            if (typeof(T) == typeof(nuint)) return (nuint)(object)left > (nuint)(object)right ? true : false;
+            if (typeof(T) == typeof(nint)) return (nint)(object)left > (nint)(object)right ? true : false;
+            if (typeof(T) == typeof(float)) return (float)(object)left > (float)(object)right ? true : false;
+            if (typeof(T) == typeof(double)) return (double)(object)left > (double)(object)right ? true : false;
+            return left.CompareTo(right) > 0 ? true : false;
+        }
+
+        public static int MoveNansToFront<TKey, TValue>(Span<TKey> keys, Span<TValue> values) where TKey : notnull
```

Comment:
Clever. :)

Reply:
@GrabYourPitchforks  Indeed, been part of .NET for a long time as `NaNPrepass` in previous native code :)